src/main/kotlin/io/libp2p/pubsub/PubsubApiImpl.kt

```diff
@@ -104,6 +104,14 @@ open class PubsubApiImpl(val router: PubsubRouter) : PubsubApi {
         return subscription
     }
 
+    override fun getPeerTopics(): CompletableFuture<Map<PeerId, Set<Topic>>> {
+        return router.getPeerTopics().thenApply { peerTopics ->
+            peerTopics.mapValues { topicNames ->
+                topicNames.value.mapTo(HashSet()) { topicName -> Topic(topicName) } as Set<Topic>
```

Comment:
```suggestion
                topicNames.value.mapTo(HashSet()) { topicName -> Topic(topicName) }
```
This cast looks obsolete here